### ISSUE TITLE
Controlling screen size not possible, especially when screen mirroring is active. `screen_deduplicate` removes "wrong" screen

### ISSUE THREAD

**Issue by MaPePeR** [awesomeWM/awesome#1927]
Output of `awesome --version`:

```
awesome v4.1 (Technologic)
 • Compiled against Lua 5.3.4 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.5
 • LGI version: 0.9.1

```

**How to reproduce the issue:**

* Dual monitor setup or external presenter connected with different resolutions.
  * For me: one laptop screen that only supports 1440x900 and a beamer that only supports  1024x768.
* Use `xrandr --output VGA-0 --mode 1024x768` or `xrandr --output VGA-0 --auto` to enable the external projector.

**Actual result:**

* Screens now overlap and [`screen_deduplicate`](https://github.com/awesomeWM/awesome/blob/master/objects/screen.c#L235-L265) will remove the smaller one.
* The smaller screen will now only show the upper left corner of the bigger screen. Every window needs to be moved and resized manually to make sure it is displayed on the other smaller accordingly.

**Expected result:**

For presentations it would be nice if i would be able to control the size of the screen myself, so i can restrict my usable awesome-screen-size to the smaller of the 2 screensizes, so i don't have to make sure that everything i see is also visible on the smaller screen by moving the windows around manually. ([`screen.geometry`](https://awesomewm.org/doc/api/classes/screen.html#screen.geometry) says it's Immutable, so i didn't try modifying that, yet)

Adding 
```
sed -E 's/first_screen->geometry.(width|height) = MAX\(first_screen->geometry.\1,\s*second_screen->geometry.\1\)/first_screen->geometry.\1 = MIN(first_screen->geometry.\1, second_screen->geometry.\1)/' -i objects/screen.c 
```
to PKGBUILD `prepare()` allowed me to - at least - invert that behaivior, but i'm not sure if that has more unforeseeable consequences and it also has some issues with not creating a blank blackground at the places where the screen-object was located before.

**Comment by Elv13** [comment]
As a workaround, you can create "fake" (virtual) screens and place them manually from Lua.

https://awesomewm.org/doc/api/classes/screen.html#screen:fake_add

or use fake resize to switch between the 2

**Comment by Elv13** [comment]
@blueyed It's not really a bug. It behave as expected by most. The `Expected result:` suggestions are also already implemented. The problem is mainly the word "fake" instead of "virtual" and the fact the `s.geometry` itself is documented as immutable, but `s:fake_resize()` mutates it anyway.

**Comment by MaPePeR** [comment]
`fake_resize` was indeed was i was looking for. I also thought it would only work on "fake screens", that were created with `fake_add`. So its mainly a documentation problem.

I'd say there is still a possible enhancement/bug left, though: The deduplication code destroys the information that is needed for finding out the positions and resolutions automatically. It probably should be part of the rc.lua, but that would require a major version bump, if i'm not mistaken.

**Comment by Elv13** [comment]
Indeed. This is a known problem and also involve other bugs such as #1344. I got a very early patch that replace the way Awesome create screens. Instead of always creating the `screen` object itself, it adds `request::area` signal to the `root` API. A Lua signal handler for this signal then creates the screen using `fake_add` and keeps track of the area. This fully remove the need for the C side deduplication and allows to move/resize existing screen screens when an external monitor is plugged in.

However this is little more than a draft and I wont have time to complete the patch anytime soon. If you wish to step in and implement the idea above, it would be welcome. It removes more code than it adds, so it is not a very complex project to undertake.
